src/adapter/graphql/tests/tests/test_gql_dataset_flow_runs.rs

```diff
@@ -196,7 +202,334 @@ async fn test_incorrect_dataset_kinds_for_flow_type() {
+
+    let res = schema
+        .execute(
+            async_graphql::Request::new(mutation_code.clone())
+                .data(harness.catalog_authorized.clone()),
+        )
+        .await;
+
+    assert!(res.is_ok(), "{:?}", res);
+    assert_eq!(
+        res.data,
+        value!({
+            "datasets": {
+                "byId": {
+                    "flows": {
+                        "runs": {
+                            "cancelFlow": {
+                                "__typename": "CancelFlowSuccess",
+                                "message": "Success",
+                                "flow": {
+                                    "__typename": "Flow",
+                                    "flowId": flow_id,
+                                    "status": "FINISHED",
+                                    "outcome": "CANCELLED"
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        })
+    );
+}
+
+////////////////////////////////////////////////////////////////////////////////////////
+
+#[test_log::test(tokio::test)]
+async fn test_cancel_transform_derived_dataset() {
+    let harness = FlowRunsHarness::new();
+    harness.create_root_dataset().await;
+    let create_derived_result = harness.create_derived_dataset().await;
+
+    let mutation_code = FlowRunsHarness::trigger_flow_mutation(
+        &create_derived_result.dataset_handle.id,
+        "EXECUTE_QUERY",
+    );
+
+    let schema = kamu_adapter_graphql::schema_quiet();
+    let res = schema
+        .execute(
+            async_graphql::Request::new(mutation_code.clone())
+                .data(harness.catalog_authorized.clone()),
+        )
+        .await;
+
+    assert!(res.is_ok(), "{:?}", res);
+    let res_json = res.data.into_json().unwrap();
+    let flow_id = res_json["datasets"]["byId"]["flows"]["runs"]["triggerFlow"]["flow"]["flowId"]
+        .as_str()
+        .unwrap();
+
+    let mutation_code =
+        FlowRunsHarness::cancel_flow_mutation(&create_derived_result.dataset_handle.id, flow_id);
+
+    let res = schema
+        .execute(
+            async_graphql::Request::new(mutation_code.clone())
+                .data(harness.catalog_authorized.clone()),
+        )
+        .await;
+
+    assert!(res.is_ok(), "{:?}", res);
+    assert_eq!(
+        res.data,
+        value!({
+            "datasets": {
+                "byId": {
+                    "flows": {
+                        "runs": {
+                            "cancelFlow": {
+                                "__typename": "CancelFlowSuccess",
+                                "message": "Success",
+                                "flow": {
+                                    "__typename": "Flow",
+                                    "flowId": flow_id,
+                                    "status": "FINISHED",
+                                    "outcome": "CANCELLED"
```

Comment:
Perhaps what's confusing me is getting
```
"status": "FINISHED",
"outcome": "CANCELLED"
```
immediately after `cancelFlow` request.

In case of a `Task` I would expect `cancel()` to just quickly set a `cancellationRequested` flag, while the task may keep running for some time.

Just trying to understand what's the actual intent here and what is the artifact of tests.

Reply:
Task will keep running, and it's result will be ignored, The immediate flow-level `CANCELLED` outcome is fine.